test/test_weakly_supervised.py

```diff
@@ -0,0 +1,214 @@
+import pytest
+from sklearn.datasets import load_iris
+from sklearn.pipeline import make_pipeline
+from sklearn.utils import shuffle, check_random_state
+from sklearn.utils.estimator_checks import is_public_parameter
+from sklearn.utils.testing import (assert_allclose_dense_sparse,
+                                   set_random_state)
+
+from metric_learn import ITML, MMC, SDML, LSML
+from metric_learn.constraints import wrap_pairs, Constraints
+from sklearn import clone
+import numpy as np
+from sklearn.model_selection import cross_val_score, train_test_split
+
+
+def build_data():
+  RNG = check_random_state(0)
+  dataset = load_iris()
+  X, y = shuffle(dataset.data, dataset.target, random_state=RNG)
+  num_constraints = 20
+  constraints = Constraints.random_subset(y)
+  pairs = constraints.positive_negative_pairs(num_constraints,
+                                              same_length=True,
+                                              random_state=RNG)
+  return X, pairs
+
+
+def build_pairs():
+  # test that you can do cross validation on a ConstrainedDataset with
```

Comment:
no ConstrainedDataset anymore. also `X_constrained` should be renamed (this is a set of pairs)

Reply:
Yes, thanks

Reply:
as some tests are parameterized to work for pairs and quadruplets, I will rename them tuples in the tests, but pairs and quadruplets in ``build_pairs`` and ``build_quadruplets`` functions that initialize data